external-resolver/src/main/scala/org/genivi/sota/resolver/vehicles/VehicleDirectives.scala

```diff
@@ -57,30 +61,68 @@ class VehicleDirectives(implicit db: Database, mat: ActorMaterializer, ec: Execu
             pathEnd {
               complete(db.run(VehicleRepository.deleteVin(vin)))
             }
-          } ~
-          pathPrefix("package") {
-            (pathEnd & get) {
-              complete(db.run(VehicleRepository.packagesOnVin(vin)))
-            } ~
-            refinedPackageId
-            { pkgId =>
-                put {
-                  complete(db.run(VehicleRepository.installPackage(vin, pkgId)))
-                } ~
-                delete {
-                  complete(db.run(VehicleRepository.uninstallPackage(vin, pkgId)))
-                }
+          }
+        } ~
+        packageRoute(vin) ~
+        componentRoute(vin)
+      }
+    }
+  }
+
+  def packageRoute(vin: Vehicle.Vin): Route = {
+    pathPrefix("package") {
+      (pathEnd & get) {
+        completeOrRecoverWith(db.run(VehicleRepository.packagesOnVin(vin))) {
+          Errors.onMissingVehicle
+        }
+      } ~
+      refinedPackageId
```

Comment:
could you please use `refinedPackageId { pkgId =>` style? Otherwise it is kind of C/C++ style